osquery/tables/events/yara_events.cpp

```diff
@@ -75,12 +75,23 @@ REGISTER(YARAEventSubscriber, "event_subscriber", "yara_events");
 Status YARAEventSubscriber::init() {
   Status status;
 
-  ConfigDataInstance config;
-  const auto& yara_config = config.getParsedData("yara");
-  if (yara_config.count("file_paths") == 0)
+  std::shared_ptr<ConfigParserPlugin> plugin;
+  try {
+    plugin = Config::getParser("yara");
```

Comment:
This interface of (1) retrieving a config parser, (2) handling exceptions and cast errors, (3) down casting again, (4) handling exceptions and cast errors again, seems like too much. Maybe we can assume that the config will only yield a `ConfigParserPlugin`, otherwise someone catastrophic has happened. Then the onerous of the caller is only on handling exceptions/errors from their single down cast.